src/main/java/rx/internal/operators/OnSubscribeLazy.java

```diff
@@ -0,0 +1,26 @@
+package rx.internal.operators;
+
+import rx.Observable;
+import rx.Subscriber;
+import rx.functions.Func0;
+
+public final class OnSubscribeLazy<T> implements Observable.OnSubscribe<T> {
+
+    final Func0<? extends T> resultFactory;
+
+    public OnSubscribeLazy(Func0<? extends T> resultFactory) {
+        this.resultFactory = resultFactory;
+    }
+
+    @Override
+    public void call(Subscriber<? super T> subscriber) {
+        try {
+            subscriber.onNext(resultFactory.call());
+        } catch (Throwable t) {
```

Comment:
Use SingleDelayedProducer, get the value  in a try-catch, set it on the producer via setValue.